src/workflow/cache_manager.rs

```diff
@@ -0,0 +1,154 @@
+use crate::workflow::WorkflowCachingPolicy;
+use lru::LruCache;
+use std::collections::HashSet;
+
+pub(crate) struct WorkflowCacheManager {
+    /// LRU cache of run IDs.
+    cache: LruCache<String, bool>,
+    /// Defines eviction behavior from the cache when tasks are completed.
+    eviction_policy: WorkflowCachingPolicy,
+    /// List of run ids that should be evicted upon completion.
+    pending_evictions: HashSet<String>,
```

Comment:
We have run id in so many places now we might want to use `Arc<String>` to avoid copying them so often but, that's a bigger change for later. Might ultimately not matter much either.

Reply:
Let's probably defer this.